src/twisted/words/protocols/irc.py

```diff
@@ -3678,10 +3678,10 @@ def ctcpExtract(message):
             normal_messages.append(messages.pop(0))
         odd = not odd
 
-    extended_messages[:] = filter(None, extended_messages)
-    normal_messages[:] = filter(None, normal_messages)
+    extended_messages[:] = list(filter(None, extended_messages))
```

Comment:
My suggestion is to have something like this

```suggestion
    extended_messages = list(filter(None, extended_messages))
```

I don't know why we need to update the list "in-place"

Reply:
@adiroiban because the two lists are inserted into the `retval` dictionary earlier. (I'm not sure if it's written that way for a reason... The whole function could probably be tidied up a lot using some list comprehensions, if someone wanted to do that.)

Reply:
Ah... I see.

so the issue is here
```
extended_messages = []
normal_messages = []
retval = {"extended": extended_messages, "normal": normal_messages}
``` 

and the return dict could have been created at the end with

```
return {"extended": extended_messages, "normal": normal_messages}
```

I think that we can live with that.